independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -18,14 +19,22 @@
 import org.eclipse.aether.RepositorySystem;
 import org.eclipse.aether.RepositorySystemSession;
 import org.eclipse.aether.repository.RemoteRepository;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.WriterConfig;
+
 import io.quarkus.bootstrap.BootstrapConstants;
 
 /**
  * Generates Quarkus extension descriptor for the runtime artifact.
  *
+ * <p/>Also generates META-INF/quarkus-extension.json which includes properties of the extension
+ * such as name, labels, maven coordinates, etc that are used by the tools.
+ *
  * @author Alexey Loubyansky
  */
-@Mojo(name = "extension-descriptor", defaultPhase = LifecyclePhase.COMPILE, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+@Mojo(name = "extension-descriptor", defaultPhase = LifecyclePhase.PROCESS_RESOURCES, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
 public class ExtensionDescriptorMojo extends AbstractMojo {
```

Comment:
is it expected that on first build you get errors like:

```
ERROR] Could not find goal 'generate-extensions-json' in plugin io.quarkus:quarkus-maven-plugin:999-SNAPSHOT among available goals add-extension, add-extensions, analyze-call-tree, build, create, create-extension, dev, generate-config, help, list-extensions, native-image, remote-dev -> [Help 1]
``` ?

second build works since then the plugin is available to use.

its as if the sequence of builds are not right.

not a biggie but just worth mentioning in case it can be avoided.

Reply:
I couldn't reproduce it neither. It might have used a locally available version of the plugin that didn't include this mojo. If you remove quarkus artifacts from your local maven repo and rebuild you shouldn't see this issue.

Reply:
i'll assume I had a bad state as I can't reproduce it neither now :)